packages/discovery-provider/plugins/pedalboard/apps/backfill-audio-analyses/src/backfill.ts

```diff
@@ -0,0 +1,223 @@
+import { AudiusLibs } from '@audius/sdk'
+import axios from 'axios';
+import { ethers } from 'ethers';
+import { Semaphore } from 'await-semaphore';
+import { Knex } from 'knex'
+import { App } from '@pedalboard/basekit'
+import { config } from '.'
+import { SharedData } from './index'
+import { storeDbOffset, readDbOffset, getCachedDiscoveryNodes, getCachedHealthyContentNodes } from './redis'
+
+// Batch size for fetching tracks
+const BACKFILL_BATCH_SIZE = 1000;
+
+// Concurrency control
+const MAX_CONCURRENT_REQUESTS = 10;
+const semaphore = new Semaphore(MAX_CONCURRENT_REQUESTS);
+
+let BACKFILL_NODES = config.environment == 'prod' 
+  ? 
+    [
+      "https://discoveryprovider.audius.co",
+      "https://discoveryprovider2.audius.co",
+      "https://discoveryprovider3.audius.co"
+    ]
+  : []
+const DEFAULT_BPM = 0
+const DEFAULT_MUSICAL_KEY = '-'
+const POLL_STATUS_INTERVAL = 5000 // 5s
+const REQUEST_TIMEOUT = 30000 // 30s
+
+interface Track {
+  track_id: number;
+  audio_upload_id: string;
+  musical_key: string | null;
+  bpm: number | null;
+  audio_analysis_error_count: number | null;
+}
+
+interface AnalysisResult {
+  audio_analysis_status: string;
+  status: string;
+  audio_analysis_results: {
+    bpm: number;
+    key: string;
+  };
+}
+
+function generateSignature(data: any, privateKey: string): string {
+  const toSignStr = JSON.stringify(data, Object.keys(data).sort());
+  const toSignHash = ethers.utils.keccak256(ethers.utils.toUtf8Bytes(toSignStr));
+  const signingKey = new ethers.utils.SigningKey(privateKey);
+  const signature = signingKey.signDigest(toSignHash);
+  return ethers.utils.joinSignature(signature);
+}
+
+function shuffleArray(array: string[]): string[] {
```

Comment:
Could we use lodash for this? https://lodash.com/docs/4.17.15#shuffle

Reply:
added in https://github.com/AudiusProject/audius-protocol/pull/8788/commits/c982665320ec617c6df43448fc879ae8c793723b